common/numerated/src/interval.rs

```diff
@@ -85,352 +67,251 @@ impl<T: Numerated> NonEmptyInterval<T> {
+        debug_assert!(start <= end, "`T: UpperBounded + LowerBounded` impl error");
+        Self { start, end }
     }
 }
 
-/// Error, which occurs, when trying to convert `start` > `end` into `Interval`.
+/// Trying to make empty interval.
 #[derive(Debug, Clone, Copy, PartialEq, Eq)]
-pub struct IntoIntervalError;
+pub struct EmptyRangeError;
 
-impl<T: Numerated, S: Into<T::B>, E: Into<T::B>> TryFrom<(S, E)> for Interval<T> {
-    type Error = IntoIntervalError;
-
-    fn try_from((start, end): (S, E)) -> Result<Self, Self::Error> {
-        use BoundValue::*;
+/// Trying to make interval from range where start > end.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub struct IncorrectRangeError;
+
+/// Trying to make interval from range where start > end or empty range.
+#[derive(Debug, Clone, Copy, PartialEq, Eq, derive_more::From)]
+pub enum IncorrectOrEmptyRangeError {
+    /// Trying to make empty interval.
+    EmptyRange(EmptyRangeError),
+    /// Trying to make interval start > end.
+    IncorrectRange(IncorrectRangeError),
+}
 
-        let start: T::B = start.into();
-        let end: T::B = end.into();
+impl<T: Numerated + UpperBounded, I: Into<T::Bound>> TryFrom<RangeFrom<I>> for Interval<T> {
+    type Error = EmptyRangeError;
 
-        match (start.unbound(), end.unbound()) {
-            (Upper(start), Upper(_)) => Ok(Self { start, end: None }),
-            (Upper(_), Value(_)) => Err(IntoIntervalError),
-            (Value(start), Upper(end)) => Ok(Self {
-                start,
-                end: Some(end),
-            }),
-            (Value(start), Value(end)) => {
-                if let Some(end) = end.dec_if_gt(start) {
-                    debug_assert!(start <= end);
-                    Ok(Self {
-                        start,
-                        end: Some(end),
-                    })
-                } else if start == end {
-                    Ok(Self { start, end: None })
-                } else {
-                    Err(IntoIntervalError)
-                }
+    fn try_from(range: RangeFrom<I>) -> Result<Self, Self::Error> {
+        match Into::<T::Bound>::into(range.start).unbound() {
+            Some(start) => {
+                let end = T::max_value();
+                debug_assert!(start <= end, "`T: UpperBounded` impl error");
+                Ok(Self { start, end })
             }
+            None => Err(EmptyRangeError),
         }
     }
 }
 
-impl<T: Numerated, I: Into<T::B>> TryFrom<Range<I>> for Interval<T> {
-    type Error = IntoIntervalError;
+impl<T: Numerated + LowerBounded + UpperBounded, I: Into<T::Bound>> TryFrom<RangeTo<I>>
+    for Interval<T>
+{
+    type Error = EmptyRangeError;
 
-    fn try_from(range: Range<I>) -> Result<Self, Self::Error> {
-        Self::try_from((range.start, range.end))
+    fn try_from(range: RangeTo<I>) -> Result<Self, Self::Error> {
+        let end: T::Bound = range.end.into();
+
+        let Some(end) = end.unbound() else {
```

Comment:
```suggestion
        let Some(end) = T::Bound::from(range.end).unbound() else {
```

Reply:
Cannot be done because T::Bound does not implement From